cmd/kubeadm/app/cmd/token.go

```diff
@@ -277,22 +296,85 @@ func RunListTokens(out io.Writer, errW io.Writer, client clientset.Interface) er
 		return errors.Wrap(err, "failed to list bootstrap tokens")
 	}
 
-	w := tabwriter.NewWriter(out, 10, 4, 3, ' ', 0)
-	fmt.Fprintln(w, "TOKEN\tTTL\tEXPIRES\tUSAGES\tDESCRIPTION\tEXTRA GROUPS")
-	for _, secret := range secrets.Items {
+	if outputFormat == "text" {
+		w := tabwriter.NewWriter(out, 10, 4, 3, ' ', 0)
+		fmt.Fprintln(w, "TOKEN\tTTL\tEXPIRES\tUSAGES\tDESCRIPTION\tEXTRA GROUPS")
+		for _, secret := range secrets.Items {
 
-		// Get the BootstrapToken struct representation from the Secret object
-		token, err := kubeadmapi.BootstrapTokenFromSecret(&secret)
-		if err != nil {
-			fmt.Fprintf(errW, "%v", err)
-			continue
+			// Get the BootstrapToken struct representation from the Secret object
+			token, err := kubeadmapi.BootstrapTokenFromSecret(&secret)
+			if err != nil {
+				fmt.Fprintf(errW, "%v", err)
+				continue
+			}
+
+			// Get the human-friendly string representation for the token
+			humanFriendlyTokenOutput := humanReadableBootstrapToken(token)
+			fmt.Fprintln(w, humanFriendlyTokenOutput)
 		}
+		w.Flush()
+	} else if outputFormat == "yaml" || outputFormat == "json" {
+		var items []*outputItem
+		for _, secret := range secrets.Items {
+			// Get the BootstrapToken struct representation from the Secret object
+			token, err := kubeadmapi.BootstrapTokenFromSecret(&secret)
+			if err != nil {
+				fmt.Fprintf(errW, "%v", err)
+				continue
+			}
+			clusterConfig, err := cmdutil.GetClusterConfig(kubeConfigFile)
+			if err != nil {
+				return errors.Wrapf(err, "failed to load cluser config from %s", kubeConfigFile)
+			}
+
+			publicKeyPins, err := cmdutil.GetCAPubKeyPins(clusterConfig)
+			if err != nil {
+				return errors.Wrapf(err, "failed to get CA certs hashes from cluster config")
+			}
 
-		// Get the human-friendly string representation for the token
-		humanFriendlyTokenOutput := humanReadableBootstrapToken(token)
-		fmt.Fprintln(w, humanFriendlyTokenOutput)
+			ttl := "<forever>"
+			expires := "<never>"
+			if token.Expires != nil {
+				ttl = duration.ShortHumanDuration(token.Expires.Sub(time.Now()))
+				expires = token.Expires.Format(time.RFC3339)
+			}
+
+			items = append(
+				items,
+				&outputItem{
+					Description:          token.Description,
+					Expires:              expires,
+					TTL:                  ttl,
+					Token:                token.Token,
+					Usages:               token.Usages,
+					Groups:               token.Groups,
+					ControlPlaneHostPort: strings.Replace(clusterConfig.Server, "https://", "", -1),
+					PublicKeyPins:        publicKeyPins,
+				},
+			)
+		}
+		var output string
+		jsn, err := json.MarshalIndent(items, "", "  ")
+		if err != nil {
+			return err
+		}
+		if outputFormat == "yaml" {
+			yml, err := yaml.JSONToYAML(jsn)
+			if err != nil {
+				return errors.Wrap(err, "failed to convert json output to yaml")
+			}
+			output = string(yml)
+		} else { //json
```

Comment:
[1]

Reply:
> I'm still wondering if it's possible to have the Yaml order the same as JSON.

sounds like the right thing to do - i.e. match the struct ordering...

it's probably some oddity between "gopkg.in/yaml.v2" and sigs.k8s.io/yaml
what happens if you marshal using `gopkg.in/yaml.v2` directly instead of converting to json first?

Reply:
Thank you. That helped. I had to use `yaml:"<capitalized field name>"` for all fields, but the order now it the same.

Changed order looks like a bug in https://github.com/go-yaml/yaml, which is copied to sigs.k8s.io/yaml.